src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -19,18 +21,21 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import jenkins.tasks.SimpleBuildStep;
 import net.sf.json.JSONObject;
 
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.StaplerRequest;
 
-public class LogParserPublisher extends Recorder implements Serializable {
+public class LogParserPublisher extends Recorder implements SimpleBuildStep, Serializable {
     private static final long serialVersionUID = 1L;
-    public final boolean unstableOnWarning;
-    public final boolean failBuildOnError;
-    public final boolean showGraphs;
-    public final String parsingRulesPath;
-    public final boolean useProjectRule;
-    public final String projectRulePath;
+    public boolean unstableOnWarning;
+    public boolean failBuildOnError;
+    public boolean showGraphs;
+    public String parsingRulesPath;

```

Comment:
I'm complaining about straightforward migration to the new approach, which causes concurrency risks in particular use-cases (see above). 

My personal opinion is that <code>@DataBoundSetter</code> is evil and requires a very accurate handling. Workflow ideally should support other ways of parameter specification.